spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1199,3 +1199,118 @@ The following metrics are produced for each channel declared by the application
 // ==== Custom message types
 
 // Implementations may provide custom subtypes of `Message` as an end user API, for the purpose of providing additional metadata associated with the messaging provider. If an application declares that it consumes or produces a subtype of the `Message` wrapper, the implementation should validate that the messaging provider supports that message wrapper type, and raise a deployment exception if it doesn't.
+
+== Publishing messages to a stream from imperative code
+Traditionally, the reactive world and imperative world are separated and operate in parallel. Reactive Messaging deals with streams of data in the reactive world, while the imperative world is pretty much point to point and synchronous communication. However, imperative programme sometimes needs to connect to reactive streams so that responses can be emitted to a destination service. 
+Bridging the two worlds is very valuable thing to do, so that one microservice can use technologies from both environments. For an instance, a JAX-RS resource might want to publish messages to a Reactive Messaging channel. This section is about enabling imperative code to publish messages to a Reactive Messaging channel, so that it can be consumed by a consumer.
+The approach is to introduce the concept of _channel_.
+
+=== Eastablish a channel
+
+In order to publish messages on a _channel_, we need to connect a channel first. `@Channel("channel_name")` is used for this purpose.
+`@Channel` is a qualifier to inject a _channel_ to a CDI bean or any class that supports CDI injection. 
+
+=== Transit messages to a channel
+
+You can either inject a _Publisher_ or _PublisherBuilder_ with the specified channel, shown below.
+
+[source, java]
+----
+
+@Inject
+@Channel("myChannel")
+private Publisher<Message<String>> field;
+
+----
+
+The value _myChannel_ indicates the name of the channel. A channel can be used for the following type:
+
+* `Publisher<Message<X>>` or `Publisher<X>`
+* `PublisherBuilder<Message<X>>` or `PublisherBuilder<X>`
```

Comment:
This section talks about injecting a `Publisher` or `PublisherBuilder` but doesn't say why you're doing this.

I assume it's so that the user can subscribe and receive messages? It might be good to spell that out. An additional example showing how to use the injected object might be helpful here too.

I'm also not sure that this actually helps bridge between reactive and non-reactive code since you still just have a reactive object. In that case, it might be better if it was moved out of the section `Publishing messages to a stream from imperative code`.

Reply:
reworked